lms/djangoapps/edraak_university/views.py

```diff
@@ -8,17 +12,45 @@
 from opaque_keys.edx.locator import CourseLocator
 from courseware.courses import get_course_with_access
 
+from courseware.access import has_access
 from student.models import UserProfile
 
 from forms import UniversityIDForm
-
+from models import UniversityID
 from helpers import get_university_id, has_valid_university_id
 
 
-class UniversityIDView(FormView):
+class CourseContextMixin(object):
+    def get_course_key(self):
+        return CourseLocator.from_string(self.kwargs['course_id'])
+
+    def get_course(self):
+        course_key = self.get_course_key()
+        return get_course_with_access(self.request.user, 'load', course_key, check_if_enrolled=True)
+
+    def require_staff_access(self):
+        course = self.get_course()
+        if not has_access(self.request.user, 'staff', course):
+            raise ObjectDoesNotExist()
```

Comment:
Well, I guess that's OK given that [edX implement something similar](https://github.com/Edraak/edx-platform/blob/master/lms/djangoapps/courseware/courses.py#L77). So instead of a `PermissionDenied` exception, they raise `UserNotEnrolled` exception which is a subclass of the `Http404` exception. Check the snapshot bellow:

![screen shot 2016-09-21 at 11 08 33 am](https://cloud.githubusercontent.com/assets/645156/18702760/c3e78864-7feb-11e6-85d3-219c67db1d9b.png)